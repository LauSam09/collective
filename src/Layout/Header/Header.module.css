.header {
  background-color: var(--colour-background);
  border-bottom: 2px solid var(--colour-action);
}

.header nav {
  padding: var(--spacing-lg);
  box-sizing: border-box;
  max-width: var(--page-width);
  width: 100%;
  margin: 0 auto;
  display: flex;
  justify-content: space-between;
}

.brand {
  margin: 0;
}

.links {
  display: flex;
  align-items: flex-end;
}

.links a {
  padding: 0 var(--spacing-sm);
}

.links a:last-of-type {
  margin-right: var(--spacing-lg);
}

.logout {
  border: none;
  background-color: inherit;
  cursor: pointer;
  padding: 0;
  color: var(--colour-action);
}

.logout:hover {
  color: var(--colour-hover);
}
