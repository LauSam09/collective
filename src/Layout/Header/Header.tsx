import { FontAwesomeIcon } from "@fortawesome/react-fontawesome"
import { faSignOutAlt } from "@fortawesome/free-solid-svg-icons"

import { useAuth, useIsAuthenticated } from "Authentication"

import classes from "./Header.module.css"

export default function Header() {
  const { logout } = useAuth()

  return (
    <header className={classes.header}>
      <nav>
        <a href="/">Collective</a>
        {useIsAuthenticated() ? (
          <button onClick={logout} title="Log out" className={classes.logout}>
            <FontAwesomeIcon icon={faSignOutAlt} size="lg" />
          </button>
        ) : null}
      </nav>
    </header>
  )
}
