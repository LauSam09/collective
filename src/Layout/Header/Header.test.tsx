import { render, screen } from "@testing-library/react"
import Header from "."

describe("<Header />", () => {
  test("renders name of application", () => {
    render(<Header />)

    expect(screen.getByText(/Collective/i)).toBeInTheDocument()
  })
})
