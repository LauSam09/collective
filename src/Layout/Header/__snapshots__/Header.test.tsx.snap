// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> matches snapshot 1`] = `
<header
  className="header"
>
  <nav>
    <a
      href="/"
      onClick={[Function]}
    >
      <span>
        <h3
          className="brand"
        >
          Collective
        </h3>
        <small>
          2.5.1
        </small>
      </span>
    </a>
  </nav>
</header>
`;
